docs/design/features/cet-feature.md

```diff
@@ -0,0 +1,28 @@
+# .NET 7 support for Intel CET shadow stacks (requires Windows)
+
+Intel�s [Control-flow Enforcement Technology (CET)](https://newsroom.intel.com/editorials/intel-cet-answers-call-protect-common-malware-threats/) is a security feature available in some newer Intel and AMD processors. It adds capabilities to the hardware that protect against some common types of attacks involving control-flow hijacking. With CET shadow stacks, the processor and operating system can track the control flow of calls and returns in a thread in the shadow stack in addition to the data stack, and detect unintended changes to the control flow. The shadow stack is protected from application code memory accesses and helps to defend against attacks involving return-oriented programming (ROP). Windows offers [Hardware-enforced Stack Protection](https://techcommunity.microsoft.com/t5/windows-kernel-internals/understanding-hardware-enforced-stack-protection/ba-p/1247815) for CET shadow stack compatibility and is available as an early preview in [Windows Insider preview builds (Dev channel)](https://insider.windows.com/en-us/understand-flighting).
+
+## Opting into CET shadow stack compatibility
+
+CET compatibility is enabled in .NET 7. In order to receive the security benefits of CET shadow stacks in .NET applications, ensure the following to opt into the feature for an executable:
+- Verify that your processor supports Intel CET. There are Intel and AMD processors currently available with the feature.
+- Get a recent build of Windows that has Hardware-enforced Stack Protection. The latest versions from [Windows Insider preview builds (Dev channel)](https://insider.windows.com/en-us/understand-flighting) have the feature.
```

Comment:
> Get a recent build of Windows that has Hardware-enforced Stack Protection.

Hardware-enforced Stack Protection support has been in the shipping Windows builds for a while.

It needs to be a build of Windows that has the final version of the APIs required to support CET with .NET. I may be worth mentioning the failure more when one uses Windows build that is too old and does not have the final version of the APIs.

Reply:
On shipping Windows builds when running on CET-enabled processors, Last I saw it was just crashing [here](https://github.com/dotnet/runtime/blob/67d2570ccfbabda2fc80b783471596d3481ffeb2/src/coreclr/vm/threads.cpp#L8216-L8217) on startup since the special user-mode APC API update to include the context was not in the shipping builds.

Reply:
I'll check if the updates have been backported.

Reply:
The updates haven't been backported yet.

Reply:
guess would be good to add details on the windows version required then.

Reply:
for now mentioned that 22H2 is required for CET.